utils/test/utils_test.cpp

```diff
@@ -347,3 +385,57 @@ TEST(util_test, test_that_verifying_a_signature_with_empty_inputs_will_fail_grac
     EXPECT_FALSE(bzn::utils::crypto::verify_signature( public_pem, "", valid_uuid));
     EXPECT_FALSE(bzn::utils::crypto::verify_signature( public_pem, signature, ""));
 }
+
+
+TEST(util_test, test_that_esr_returns_peers_list)
+{
+    const std::pair<bzn::uuid_t, size_t> BAD_SWARM{"NonExistentBluzelleSwarm", 0};
+
+    // Check for swarm that doesn't exist, must return empty vector.
+    {
+        const auto peer_ids = bzn::utils::esr::get_peer_ids(BAD_SWARM.first);
+        EXPECT_EQ( BAD_SWARM.second, peer_ids.size());
+    }
+
+    // There are three test swarms
+    for (const auto& swarm : SWARMS)
+    {
+        std::vector<bzn::uuid_t> accepted_ids(swarm.second.size());
+        std::transform(swarm.second.begin(), swarm.second.end(), accepted_ids.begin(), [](const auto &pinfo){return pinfo.uuid;});
+        const auto peer_ids = bzn::utils::esr::get_peer_ids(swarm.first);
+        EXPECT_EQ(swarm.second.size(), accepted_ids.size());
+        EXPECT_EQ(accepted_ids, peer_ids);
+    }
+}
+
+
+TEST(util_test, test_that_esr_returns_peer_info)
+{
+    {
+        const auto peer_info{bzn::utils::esr::get_peer_info("BluzelleSwarm", "NotAGoodID")};
+        EXPECT_EQ(bzn::peer_address_t("", 0, 0, "", "NotAGoodID"), peer_info);
+    }
+
+    {
+        const auto peer_info{bzn::utils::esr::get_peer_info("NotABluzelleSwarm", "NotAGoodID")};
+        EXPECT_EQ(bzn::peer_address_t("", 0, 0, "", "NotAGoodID"), peer_info);
+    }
+
+    // exhaustive testing must occur in integration testing, since this functionality takes a long time we will randomly
+    // select a single swarm and a single node to test against.
+    {
+        std::default_random_engine generator;
+        generator.seed(time(nullptr));
```

Comment:
this will give me no end of grief with valgrind again. Can't you just pick 1 entry instead?
What are you gaining by making this random?

Reply:
I started with one example of a swarm and parsed that one, when Matthew created other examples for me my first attempt at a parser wasn't general enough and he helped me discover the deficiencies in my assumptions about output data.

I'll make the change though.

Reply:
Done, removed the randomness.